### PR TITLE
refactor!: Rename Seedfinder to SeedFinder - classes and files

### PR DESCRIPTION
**Pull request by tboldagh**
The Seedfinder and related classes somehow did not follow the usual naming convention for classes (CamelCase convention). This PR fixes this. 

It addresses part of the issue: #1522 but not closes it. I decided that is best addressed in two PRs. This one is strictly technical.

BREAKING CHANGE: Include file names are changed and so are the classes. Fix is trivial on client side, yet needs to be done in several places when migrating.

### CODE REVIEW

File: Examples/Python/python/acts/examples/reconstruction.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def addSeeding(
     truthSeedRanges: Optional[TruthSeedRanges] = TruthSeedRanges(),
     particleSmearingSigmas: ParticleSmearingSigmas = ParticleSmearingSigmas(),
     initialVarInflation: Optional[list] = None,
-    seedfinderConfigArg: SeedfinderConfigArg = SeedfinderConfigArg(),
+    seedfinderConfigArg: SeedFinderConfigArg = SeedFinderConfigArg(),
```
Comment (LuisFelipeCoelho) [inline Examples/Python/python/acts/examples/reconstruction.py:151]: should we change seedfinderConfigArg to seedFinderConfigArg as well?

**Review by LuisFelipeCoelho — COMMENTED** [review]
Thanks a lot for this PR

**Comment by tboldagh** [comment]
The changes here are trivial but in many files. Probably a good idea would be to merge it soon to avoid conflicts. @asalzburger ?

**Review by AJPfleger — COMMENTED** [review]
The changes look good to me. I want to leave the final decision to @asalzburger since there are changes on the client side.

**Comment by tboldagh** [comment]
I have some git (understanding) issues that prohibit me from pushing the fix.

**Comment by tboldagh** [comment]
@asalzburger - since you were assigned as the final judge can you let me know when this PR has chance to be merged. I can keep fixing conflicts but if there is not chance for the merge any time soon I'll spare few cycles of it.

**Review by asalzburger — APPROVED** [review]

**Comment by asalzburger** [comment]
Unfortunately there are some conflicts, once you can resolve them, we go ahead.

**Comment by tboldagh** [comment]
> Unfortunately there are some conflicts, once you can resolve them, we go ahead.

@asalzburger - the conflicts are gone.

**Review by paulgessinger — APPROVED** [review]

**Comment by paulgessinger** [comment]
Ok let's just get this in!

**Pull request merged**